### ISSUE TITLE
Loss function of LogisticRegressionOutput

### ISSUE THREAD

**Issue by marugari** [dmlc/mxnet#3355]
It seems that LogisticRegressionOutput has squared loss function (and logit link function).
Could you confirm my understanding is correct?
https://github.com/dmlc/mxnet/blob/ce92c1074754a5a5fb607d3c59dd119c37cd6aea/src/operator/regression_output.cu#L19

https://github.com/dmlc/mxnet/issues/784#issuecomment-161472054
https://github.com/dmlc/MXNet.jl/issues/67#issuecomment-203815088

**Comment by pluskid** [comment]
It seems to be inconsistent with the conventional "logistic regression". The real logistic regression should be the `SoftmaxOutput`.

**Comment by marugari** [comment]
I got it.

However, For binary classification I want to use ``LogisticRegressionOutput``, because it reduce ``num_hidden`` to 1. (It decreases number of weights to be estimated?)

**Comment by marugari** [comment]
It has already been discussed.
https://github.com/dmlc/mxnet/issues/2001
